Refactor : Simplify usage of platform by getting rid of platform in the store

dgengler6 opened the pull request:
Fixes #262 #243

**What this does ?**
- Remove the Federated/Decentralized split in platform
- Remove platform from the store and leaves it up to the training manager 
- Added both `Train Federated` and `Train Decentralized` buttons to the training frame
- The client is created when one of these options is chosen
- Remove platform switch button in settings
- Create Locales files, for now there is only english but now we are open to translation
- It would be great to integrate all text of the app in the I18n.json and keep this as a good practice
- Change platform I18n to an actual locale manager for later translations ( One could create a french.json file, copy pasting default.json (english locale) and translate the texts )

**TODO steps**

- [x] Training platform is not dependent on the store anymore
- [x] Adapt Training frame so that we can chose between fed/dec
- [x] Remove platform completely from the store
- [x] Adapt I18n content so they fit the hybrid platform
- [x] Add scheme field to the task
- [x] Create english locale and lay the base for translation

dgengler6 committed:
Start to remove references to platform in the store

dgengler6 commented:
There is still a reference to the platform in the store because it is used for TaskCreationForm, we can change this I am just not sure what is the use of that there.

dgengler6 committed:
Fix tests with new TrainingManager constructor

martinjaggi commented:
thanks for doing this, this will make things simpler indeed.

while you're already doing it, you could simplify even more, we actually don't need any UI element.
you could just add a boolean field to the task config (each task will be predefined either federated or decentralized). 

the reason is that we don't want to allow that people join the same task in potentially different modes. if both modes are wanted people should create 2 separate tasks, one federated one decentralized

Nacho114 commented:
> thanks for doing this, this will make things simpler indeed.
> 
> while you're already doing it, you could simplify even more, we actually don't need any UI element. you could just add a boolean field to the task config (each task will be predefined either federated or decentralized).
> 
> the reason is that we don't want to allow that people join the same task in potentially different modes. if both modes are wanted people should create 2 separate tasks, one federated one decentralized

I agree, but there are different ways to implement this, might be good to brainstorm together. The training scheme choice is an important part of Disco so it's good to spend some time discussing on how it is being offered to the user.

dgengler6 committed:
Remove I18n federated.json and decentralized.json, change platform.config.js to locales.config.js, set english as default

dgengler6 commented:
So last push changed quite a lot. 

- Now we only have one I18n (i.e default.json) which actually corresponds to the english locale
- the platform directory is now called locales for obvious reasons
- I renamed trainingType into trainingSchemes and moved it in the training folder
- Fully removed any reference to store.getters.platform

dgengler6 committed:
make the test actually work

Nacho114 commented:
You can also add in the description that it fixes #243, that issue was in regards to cleaning i18n which is done here.

Nacho114 commented on mobile-browser-based-version/src/components/building_frames/containers/TrainingFrame.vue:
```diff
@@ -21,15 +21,21 @@
           <custom-button
             :center="true"
-            @click="training_manager.joinTraining(true)"
+            @click="training_manager.joinTraining(decentralizedScheme)"
```
@s314cy @dgengler6  maybe we could add:

joinFederatedTraining(), joinDecentralizedTraining(), joinLocalTraining(), into the training manager, that way we don't pass arguments and it's resolved in the training_manager?

dgengler6 committed:
create training functions for each scheme, add texts to I18n

dgengler6 committed:
add scheme field to task's trainingInformation

martinjaggi commented:
thanks!
BTW the vue components where we join training maybe don't even need to know if it's federated or not.

it could be figured out later in `training_manager`. so instead of `training_manager.joinFederatedTraining()` one could just make a `training_manager.joinTraining()` and that one would then look at the task config so it would know what to join (federated or decentralized). training alone is different and has it's own button as usual

Nacho114 commented:
> thanks! BTW the vue components where we join training maybe don't even need to know if it's federated or not.
> 
> it could be figured out later in `training_manager`. so instead of `training_manager.joinFederatedTraining()` one could just make a `training_manager.joinTraining()` and that one would then look at the task config so it would know what to join (federated or decentralized). training alone is different and has it's own button as usual

yeah i agree, each component should know as little as possible 👍🏼

dgengler6 committed:
There is now only one button for training, the scheme used is defined in trainingManager and depends on task.trainingInformation.scheme

dgengler6 commented:
Done, Although I think we should be clear that we should be clear about what is happening under the hood somewhere (i.e the toast alerts) 
Tbh I don't know if having two times the same task, but one Federated and one Decentralized makes much sense, it might just be confusing for the user..

Nacho114 commented:
> Done, Although I think we should be clear that we should be clear about what is happening under the hood somewhere (i.e the toast alerts) Tbh I don't know if having two times the same task, but one Federated and one Decentralized makes much sense, it might just be confusing for the user..

hmm yeah, UI design is not obvious, we can have a discussion on Tuesday -- but keeping it simple is a good rule of thumb, although what simple means is what's hard to figure out

dgengler6 committed:
merge and adapted all files, ready for testing

dgengler6 committed:
remove duplicated files

dgengler6 committed:
fix the tests, make disco's connect/disconnect and init client private

dgengler6 committed:
small refactor and fix client init bug

Nacho114 committed:
Merge branch 'develop' into 262-simplify-platform-usage

dgengler6 committed:
merged

dgengler6 committed:
fix tests, fix fed/dec UI training information

dgengler6 committed:
fixed merge to remove platform

dgengler6 committed:
Resolved discojs export error

dgengler6 committed:
remove unused import in discojs index

dgengler6 commented:
I think I managed to merge everything with the new architecture ! 

Some tests don't pass but I'll fix that asap, It might have smthg to do with the training errors we had.

s314cy commented:
looking really nice! on a side note, the `activePage` store has been removed on the current develop version, as the components now communicate between each other instead

would you like to meet with @Nacho114 and me to ensure the CI tests are successful on your next commit?

dgengler6 commented:
> looking really nice! on a side note, the `activePage` store has been removed on the current develop version, as the components now communicate between each other instead
> 
> would you like to meet with @Nacho114 and me to ensure the CI tests are successful on your next commit?

Yeah we can meet a bit later this afternoon !

dgengler6 committed:
done merging

dgengler6 committed:
test

dgengler6 committed:
CI tests are now working

Nacho114 reviewed (changes requested):
Nice work :)

I tried it out but there is an issue with the stop button not appearing after starting training. Once this is fix we will rebase it with devdelop!

dgengler6 committed:
fix isTraining set to true after training only

dgengler6 committed:
Merge branch 'develop' into 262-simplify-platform-usage

dgengler6 committed:
merge latest PR

dgengler6 committed:
remove call to platform in benchmark

Nacho114 reviewed (approved):
Great simplification, good work :)

martinjaggi commented:
thanks a lot @dgengler6 !
